### ISSUE TITLE
Empty string in workspace_key_prefix breaks workspace functionality

### ISSUE THREAD

**Issue by hackdna** [hashicorp/terraform#16387]
### Terraform Version
```
$ terraform -v
Terraform v0.10.7
```

### Terraform Configuration Files
```hcl
terraform {
  required_version = "~> 0.10"

  backend "s3" {
    key                  = "storage/terraform.tfstate"
    workspace_key_prefix = ""
  }
}
```

### Expected Behavior
Non-default workspaces should be usable.

### Actual Behavior
```
Workspace "test-workspace" doesn't exist.

You can create this workspace with the "new" subcommand.
```

### Steps to Reproduce
1. `terraform init`
1. `terraform workspace new test-workspace`
1. `terraform workspace select default`
1. `terraform workspace select test-workspace`

### References
May be related to #16383

**Comment by sunil592002** [comment]
Hi,

I am also facing the same issue. Is this bug will be fixed soon, to make us of workspace in PROD

**Comment by jbardin** [comment]
closed via #16932
